core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1078,19 +1015,19 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
                       accessLevel,
                       workspaceDetails,
                       submissionStats,
-                      workspaceSamResource.public.roles.nonEmpty || workspaceSamResource.public.actions.nonEmpty
+                      workspaceSamResource.public.roles.nonEmpty || workspaceSamResource.public.actions.nonEmpty,
+                      Some(wsmContext.policies)
                     )
                   )
                 } catch {
                   // Internal folks may create MCWorkspaces in local WorkspaceManager instances, and those will not
                   // be reachable when running against the dev environment.
-                  case ex: ApiException
-                      if (WorkspaceType.McWorkspace == workspace.workspaceType) && (ex.getCode == StatusCodes.NotFound.intValue) =>
+                  case ex: AggregateWorkspaceNotFoundException =>
                     logger.warn(
                       s"MC Workspace ${workspace.name} (${workspace.workspaceIdAsUUID}) does not exist in the current WSM instance. "
                     )
                     None
-                  case ex: InvalidCloudContextException =>
+                  case ex: WorkspaceAggregationException =>
```

Comment:
@aherbst-broad @blakery I think we lost some functionality here introduced in https://github.com/broadinstitute/rawls/pull/2464-- this was handling a missing CloudContext during the list to avoid throwing an exception. As far as I can tell, with the refactoring in this PR we will still get an InvalidCloudContextException from AggregatedWorkspaceService, but it will no longer be handled here.

Reply:
Though actually the unit test still existed-- https://github.com/broadinstitute/rawls/pull/2491/files#diff-05872226f80cea9f996d66c8df13e83d05cf72a93a2fec9fa8d5a3e54550f32eR3300. I will try running the test tomorrow and see how this is still being handled.

Reply:
Ah, I see. The `InvalidCloudContextException` ends up getting wrapped in a `WorkspaceAggregationException`.